sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -1106,6 +1106,16 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-42286: Insert into a table select from case when with cast, positive test") {
```

Comment:
Have you tried running the test without the code change?

Reply:
Before this change, we would hit 
```
java.lang.ClassCastException: class org.apache.spark.sql.catalyst.expressions.CaseWhen cannot be cast to class org.apache.spark.sql.catalyst.expressions.AnsiCast
```